docs/serviceability.md

```diff
@@ -0,0 +1,76 @@
+```
+
+### Deployed on a Managed Cluster
+
+To access the application-service controller logs on either the AppStudio Staging or Production clusters, you would need access to CloudWatch service. To learn more about how to request access to CloudWatch and view the application-service logs, refer to the HAS Access Control [documentation](https://docs.google.com/document/d/1cK4XGKpXBEYOKfIqSiHuuCfsfHjElxhG9lrlEozzgVE/edit#heading=h.yxk6h5uvh57d). 
+
+## Understanding the Logs
+Each application-service controller logs their reconcile logic to the manager. The log message format is generally of format 
+
+```
+{"level":"info","ts":"2023-08-31T19:59:21.144Z","msg":"Finished reconcile loop for user-tenant/devfile-sample-go-basic-development-binding-hr9nm","controller":"snapshotenvironmentbinding","controllerGroup":"appstudio.redhat.com","controllerKind":"SnapshotEnvironmentBinding","SnapshotEnvironmentBinding":{"name":"devfile-sample-go-basic-development-binding-hr9nm","namespace":"user-tenant"},"namespace":"user-tenant","name":"devfile-sample-go-basic-development-binding-hr9nm","reconcileID":"d5c8545b-957b-4f1a-b177-84a5d5f0d26c"}
+```
+
+To understand the AppStudio controller logging convention, refer to the Appstudio [ADR](https://github.com/redhat-appstudio/book/blob/main/ADR/0006-log-conventions.md)
+
+## Debugging
+
+- Insert break points at the controller functions to debug unit tests
+- When debugging unit and integration tests, remember that the mock clients used are hosting dummy data and mocking the api call and returns
+- For debugging controller logs either from a local cluster or a managed cluster, gather the log and search for:
+  - the namespace the user belongs to. For example, if the user is on the project/namespace, search for `user-tenant`
+  - the resource name that the controller is reconciling. For example, if the `SnapshotEnvironmentBinding` resource is being reconciled, then search for `"name":"devfile-sample-go-basic-development-binding-hr9nm"` belonging to `"controllerKind":"SnapshotEnvironmentBinding"`
+  - the log message. For example, `"msg":"Finished reconcile loop for user-tenant/devfile-sample-go-basic-development-binding-hr9nm"`. You may search for the string `Finished reconcile loop for` in the application-service repository to track down the code logic that is emitting the log. Remember to look out for resource name concatenation and/or error wrapping that may not turn up in your code search. It is advised to exclude such strings from the code search for debugging purposes
+
+## Common Problems
+- When deploying HAS locally or on a local cluster, a Github Personal Access Token is required as the application-service controller requires the token for pushing the resources to the GitOps repository. Please refer to the [instructions](../docs/deploy.md#creating-a-github-secret-for-application-service) in the deploy section for more information
```

Comment:
I don’t think I saw steps for deploying HAS locally (I.e. outside of a container)?